### ISSUE TITLE
[BUG] hash down long DepIDs

### ISSUE THREAD

**Issue by isaacs** [vltpkg/vltpkg#85]
There are cases where a `DepID` can get very very long.

For example, if a package has a dependency on a local tarball, like `{ "dependencies": { "a" : "file:./a.tgz" }}`, then the resulting DepID in the graph will be something like:

```
file;${encode('node_modules/.vlt/registry;;pkg@1.2.3')}
```

If that tarball then does the same thing, say `a.tgz` has a `package.json` wiht `{ "dependencies": { "b": "file:./b.tgz" }}`, then it'll have a DepID like:

```
file;${encode('node_modules/.vlt/file;${encode('node_modules/.vlt/registry;;pkg@1.2.3')}`)}
```

Nest 3 or 4 deep, and you start blowing out path length limits.

## proposal

If a DepID is longer than a certain number of characters (like, idk, 100 or so?) hash and truncate.

So, this:

```
file;${encode('node_modules/.vlt/file;${encode('node_modules/.vlt/registry;;pkg@1.2.3')}`)}
```

becomes:

```
const MAX = 100
const HASHLEN = 10
const s = `file;${encode('node_modules/.vlt/file;${encode('node_modules/.vlt/registry;;pkg@1.2.3')}`)}`
return s.length < MAX ? s : s.substring(0, MAX - HASHLEN) + hash(s).substring(0, HASHLEN)
```

**Comment by darcyclarke** [comment]
@isaacs / @ruyadorno is this still a problem?

**Comment by ruyadorno** [comment]
yes, this is still an architecture challenge that we'll have to tackle at some point, particularly ahead of starting the work to support peer deps

**Comment by darcyclarke** [comment]
Closing as this got resolve for peer deps & we can open a new issue for modifiers or other, similar needs in the future
